networkx/readwrite/nx_shp.py

```diff
@@ -73,8 +78,12 @@ def read_shp(path):
                 attributes["Wkb"] = g.ExportToWkb()
                 attributes["Wkt"] = g.ExportToWkt()
                 attributes["Json"] = g.ExportToJson()
-                last = g.GetPointCount() - 1
-                net.add_edge(g.GetPoint_2D(0), g.GetPoint_2D(last), attributes)
+                if simplify:
+                    last = g.GetPointCount() - 1
+                    net.add_edge(g.GetPoint_2D(0), g.GetPoint_2D(last), attributes)
+                else:
+                    for i in range(0, g.GetPointCount() - 1):
```

Comment:
The `last` variable should be declared after line 80 where it previously was. Then line 85 should be changed to 
```python
for i in range(last):

Don't understand why variable `last` is inside the `if` block.